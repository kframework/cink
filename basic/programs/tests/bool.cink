int main() {
  int a = 1, b = 2, c = 3;
  if ( a < 2 && b >1 || c > 4) 
    cout << "true";
  else cout << "false";
 cout << endl;
  return 0;
}

