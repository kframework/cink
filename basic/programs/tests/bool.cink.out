true
<T>
    <in>
        ListItem(#buffer ( "" ))
        ListItem(#istream ( 0 ))
    </in>
    <out>
        ListItem(#ostream ( 1 ))
        ListItem(#buffer ( "" ))
    </out>
    <control>
        <k>
            .K
        </k>
        <return>
            0
        </return>
        <fstack>
            .List
        </fstack>
        <sideEffects>
            .Bag
        </sideEffects>
        <env>
            main |-> #symScalLoc(1)
        </env>
    </control>
    <genv>
        main |-> #symScalLoc(1)
    </genv>
    <store>
        #symScalLoc(1) |-> lambda ( .Decls @ (((((((int a) ;) ((a = 1) ;))
           ((((int b) ;) ((b = 2) ;)) (((int c) ;) ((c = 3) ;)))) (if ( (((a <
           2) && (b > 1)) || (c > 4)) ) ((cout << "true") ;) else ((cout << 
          "false") ;))) ((cout << endl) ;)) (return 0 ;)) )
        #symScalLoc(2) |-> 1
        #symScalLoc(3) |-> 2
        #symScalLoc(4) |-> 3
    </store>
</T>
